src/Hl7.Fhir.Core/Rest/FhirClient.cs

```diff
@@ -19,32 +19,44 @@ namespace Hl7.Fhir.Rest
 {
     public partial class FhirClient : BaseFhirClient
     {
-//disables warning that OnBeforeRequest and OnAfterResponse are never used.
+        //disables warning that OnBeforeRequest and OnAfterResponse are never used.
 #pragma warning disable CS0067
 
         /// <summary>
         /// Creates a new client using a default endpoint
         /// If the endpoint does not end with a slash (/), it will be added.
         /// </summary>
         /// <remarks>
-        /// If the messageHandler is provided then it must be disposed by the caller
+        /// If the messageHandler, or httpClient is provided then it must be disposed by the caller
+        /// Only one of the messageHandler or httpClient can be provided
         /// </remarks>
         /// <param name="endpoint">
         /// The URL of the server to connect to.<br/>
         /// If the trailing '/' is not present, then it will be appended automatically
         /// </param>
         /// <param name="settings"></param>
         /// <param name="messageHandler"></param>
+        /// <param name="httpClient"></param>
         /// <param name="provider"></param>
-        public FhirClient(Uri endpoint, FhirClientSettings settings = null, HttpMessageHandler messageHandler = null, IStructureDefinitionSummaryProvider provider = null) : base(endpoint, settings, provider)
+        public FhirClient(Uri endpoint, FhirClientSettings settings = null, HttpMessageHandler messageHandler = null, HttpClient httpClient = null, IStructureDefinitionSummaryProvider provider = null) : base(endpoint, settings, provider)
```

Comment:
I think it is better to create an overload of the constructor with only the `HttpClient` as a parameter and not both (`HttpMessageHandler` and  `HttpClient`). Then the check for on line 46 is also not needed.

```c#
public FhirClient(Uri endpoint, FhirClientSettings settings = null, HttpClient httpClient = null, IStructureDefinitionSummaryProvider provider = null) : base(endpoint, settings, provider)
{
 // etc.
}
```

Of couse you need the other overload with Uri as a string as well:
```c#
 public FhirClient(string endpoint, FhirClientSettings settings = null, HttpClient httpClient = null, IStructureDefinitionSummaryProvider provider = null)
            : this(new Uri(endpoint), settings, httpClient, provider)
        {
        }
```

Reply:
2 constructors to 4 is why I did like this, but no drama, can switch it over pretty quick and get it back in.